enhancements/sig-architecture/32-extensiblescheduling/32-extensiblescheduling.md

```diff
@@ -0,0 +1,363 @@
+  - On the hub, there is a customized agent to get metrics (eg. cluster allocatable memory) from Thanos. It will generate a scalar value for each cluster.
+  - As an end user, I can configure placement yaml to use this scalar value to select clusters.
+
+#### Story 3: Disaster recovery workload could be automatically switched to an avaliable cluster.
+  - A user has two clusters, a primary and backup cluster on which workload storage is configured so storage are synced from primary cluster to backup.
+  - As an end user, I want to deploy workloads on the primary cluster first. And when primary cluster is down, the workload should be automatically switched to the backup cluster.
+
+#### Story 4: Users could use the data provided by third party controllers to select clusters.
+  - A third party controller can comprehensively evaluate cluster on latency, region, iops etc. and rate cluster according to the SLA the cluster can provide.
+  - As an end user, I can configure placement yaml to use this scalar value to select clusters.
+
+### Risks and Mitigation
+N/A
+
+## Design Details
+
+### ManagedClusterScalar API
+ManagedClusterScalar is the new API to add, to represents a scalable value (aka score) of one managed cluster.
+```go
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope="Namespaced"
+// +kubebuilder:subresource:status
+
+// ManagedClusterScalar represents a scalable value (aka score) of one managed cluster.
+// Each ManagedClusterScalar only represents the scalar value of one prioritizer.
+// ManagedClusterScalar is a namesapce scoped resource. The namespace of the resource is the cluster namespace.
+type ManagedClusterScalar struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the attributes of the ManagedClusterScalar.
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ManagedClusterScalarSpec `json:"spec"`
+
+	// Status represents the status of the ManagedClusterScalar.
+	// +optional
+	Status ManagedClusterScalarStatus `json:"status,omitempty"`
+}
+
+// ManagedClusterScalarSpec defines the attributes of the ManagedClusterScalar.
+type ManagedClusterScalarSpec struct {
+	// PrioritizerName will be the prioritizer name used in placement.
+	// Each ManagedClusterScalar only represent the scalar value of one prioritizer for one cluster.
+	// +kubebuilder:validation:Required
+	// +required
+	PrioritizerName string `json:"prioritizerName,omitempty"`
+}
+
+//ManagedClusterScalarStatus represents the current status of ManagedClusterScalar.
+type ManagedClusterScalarStatus struct {
+	// Conditions contain the different condition statuses for this managed cluster scalar.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions"`
+
+	// Scalar contains a scalable value of this managed cluster.
+	// +optional
+	Scalar int64 `json:"scalar,omitempty"`
```

Comment:
to be weighted, we wanted to force a range, rigth?  -100/+100 is what I recall.

Reply:
-100/+100 indeed makes the user easy to understand the value, while the idea in this proposal is not to set the scalar range (placement-controller prioritizer plugin will range the final score to -100/+100)

The reason is, if the controller is running on the managed cluster, it doesn't have a global view to normalize the scalar into -100/+100. Or needs to set an appropriate default min/max value to normalize the scalar, the side effect is it will lose resolution. 

What will you suggest about this?